5calls/app/src/androidTest/java/org/a5calls/android/a5calls/controller/MainActivityErrorTest.java

```diff
@@ -0,0 +1,199 @@
+
+        // Set a mock location to avoid location prompts
+        AccountManager.Instance.setAddress(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                "90210");
+
+        // Mark tutorial as seen to bypass onboarding screen
+        AccountManager.Instance.setTutorialSeen(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                true);
+
+        // Create mock HTTP stack
+        mHttpStack = new MockHttpStack();
+    }
+
+    @After
+    public void tearDown() {
+        // Restore original state
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setRequestQueue(mOriginalRequestQueue);
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setFiveCallsApi(mOriginalApi);
+
+        // Restore original location
+        AccountManager.Instance.setAddress(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                mOriginalAddress);
+
+        // Close the activity scenario if it's open
+        if (scenario != null) {
+            scenario.close();
+        }
+    }
+
+    /**
+     * Sets up the mock request queue and API
+     */
+    private void setupMockRequestQueue() {
+        // Create a custom RequestQueue with our mock HTTP stack
+        BasicNetwork basicNetwork = new BasicNetwork(mHttpStack);
+        FakeRequestQueue requestQueue = new FakeRequestQueue(basicNetwork);
+        requestQueue.start();
+
+        // Replace the app's RequestQueue with our mock
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setRequestQueue(requestQueue);
+
+        // Create a new FiveCallsApi with our mock RequestQueue
+        String callerId = AccountManager.Instance.getCallerID(
+                InstrumentationRegistry.getInstrumentation().getTargetContext());
+        FiveCallsApi api = new FiveCallsApi(callerId, requestQueue);
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setFiveCallsApi(api);
+    }
+
+    /**
+     * Launches the MainActivity and waits for it to load
+     * @param waitTimeMs time to wait for the activity to load and process errors
+     */
+    private void launchMainActivity(int waitTimeMs) {
+        // Launch the activity
+        scenario = ActivityScenario.launch(MainActivity.class);
+
+        // Wait for error processing and UI to update
+        try {
+            Thread.sleep(waitTimeMs);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Custom matcher to check if a RecyclerView has at least one item
+     */
+    public static Matcher<View> hasAtLeastOneItem() {
```

Comment:
in the error state the recycler view should have exactly one item. can you change this to hasExactlyOneItem and getItemCount == 1?

Reply:
done